### ISSUE TITLE
[BUG] Error in <Anonymous>: unable to find an inherited method for function ‘.showSelectionDetails’ for signature ‘"CustomTable"’

### ISSUE THREAD

**Issue by kniekamp19** [iSEE/iSEE#608]
Hello,

When creating a custom table in a fairly basic iSEE app, an error is printed under the table stating " **Error:** unable to find an inherited method for function ‘.showSelectionDetails’ for signature ‘"CustomTable"’ ". This remains whether I use my SingleCellExperiment data or a simple artificial dataframe (see below). The table still prints as expected, but the error underneath is going to cause confusion when I publish the app for my team to use. No amount of error handling/message suppression has removed the error message from the app. This is on Windows 10, sessionInfo() is pasted at the bottom. Thanks for any help/guidance.

**To Reproduce**
TEST_TABLE <- function(se, ri=row.names(colData(se)),ci) {
  data.frame(
    x = 1:10,
    y = letters[1:10],
    row.names=paste(letters[1:10],1:10, sep='')
  )
}

TESTTBLGEN <- createCustomTable(TEST_TABLE)
test.tbl <- TESTTBLGEN()

iSEE(se, initial=list(test.tbl))

![image](https://user-images.githubusercontent.com/55200737/202027911-542d37b8-4079-4274-a7b7-01a150b0b66b.png)

**Session info (please complete the following information)**
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tradeSeq_1.12.0             ggbeeswarm_0.6.0            SeuratObject_4.1.3          Seurat_4.2.1               
 [5] ggplot2_3.4.0               shiny_1.7.3                 iSEE_2.10.0                 SingleCellExperiment_1.20.0
 [9] SummarizedExperiment_1.28.0 Biobase_2.58.0              GenomicRanges_1.49.0        GenomeInfoDb_1.34.3        
[13] IRanges_2.32.0              S4Vectors_0.36.0            BiocGenerics_0.44.0         MatrixGenerics_1.10.0      
[17] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             shinydashboard_0.7.2   spatstat.explore_3.0-5 reticulate_1.26        tidyselect_1.2.0      
  [6] htmlwidgets_1.5.4      grid_4.2.1             BiocParallel_1.32.1    Rtsne_0.16             munsell_0.5.0         
 [11] codetools_0.2-18       ragg_1.2.4             ica_1.0-3              DT_0.26                future_1.29.0         
 [16] miniUI_0.1.1.1         withr_2.5.0            spatstat.random_3.0-1  colorspace_2.0-3       progressr_0.11.0      
 [21] knitr_1.40             ROCR_1.0-11            tensor_1.5             shinyWidgets_0.7.4     listenv_0.8.0         
 [26] labeling_0.4.2         GenomeInfoDbData_1.2.9 polyclip_1.10-4        farver_2.1.1           parallelly_1.32.1     
 [31] vctrs_0.5.0            generics_0.1.3         xfun_0.34              R6_2.5.1               doParallel_1.0.17     
 [36] clue_0.3-62            locfit_1.5-9.6         bitops_1.0-7           spatstat.utils_3.0-1   cachem_1.0.6          
 [41] shinyAce_0.4.2         DelayedArray_0.23.2    assertthat_0.2.1       promises_1.2.0.1       scales_1.2.1          
 [46] beeswarm_0.4.0         gtable_0.3.1           globals_0.16.1         goftest_1.2-3          rlang_1.0.6           
 [51] systemfonts_1.0.4      GlobalOptions_0.1.2    splines_4.2.1          lazyeval_0.2.2         princurve_2.1.6       
 [56] spatstat.geom_3.0-3    yaml_2.3.5             reshape2_1.4.4         abind_1.4-5            crosstalk_1.2.0       
 [61] httpuv_1.6.6           tools_4.2.1            ellipsis_0.3.2         jquerylib_0.1.4        RColorBrewer_1.1-3    
 [66] ggridges_0.5.4         Rcpp_1.0.9             plyr_1.8.8             zlibbioc_1.44.0        purrr_0.3.5           
 [71] RCurl_1.98-1.9         TrajectoryUtils_1.6.0  deldir_1.0-6           pbapply_1.5-0          GetoptLong_1.0.5      
 [76] viridis_0.6.2          cowplot_1.1.1          fontawesome_0.4.0      zoo_1.8-11             ggrepel_0.9.2         
 [81] cluster_2.1.4          magrittr_2.0.3         data.table_1.14.4      scattermore_0.8        circlize_0.4.15       
 [86] colourpicker_1.2.0     lmtest_0.9-40          RANN_2.6.1             fitdistrplus_1.1-8     patchwork_1.1.2       
 [91] shinyjs_2.1.0          mime_0.12              evaluate_0.18          xtable_1.8-4           gridExtra_2.3         
 [96] shape_1.4.6            compiler_4.2.1         tibble_3.1.8           KernSmooth_2.23-20     crayon_1.5.2          
[101] htmltools_0.5.3        mgcv_1.8-41            later_1.3.0            tidyr_1.2.1            DBI_1.1.3             
[106] slingshot_2.6.0        ComplexHeatmap_2.14.0  MASS_7.3-58.1          Matrix_1.5-3           cli_3.4.1             
[111] parallel_4.2.1         igraph_1.3.5           pkgconfig_2.0.3        sp_1.5-1               plotly_4.10.1         
[116] spatstat.sparse_3.0-0  foreach_1.5.2          vipor_0.4.5            bslib_0.4.1            XVector_0.38.0        
[121] stringr_1.4.1          digest_0.6.29          sctransform_0.3.5      RcppAnnoy_0.0.20       spatstat.data_3.0-0   
[126] rmarkdown_2.18         leiden_0.4.3           rintrojs_0.3.2         uwot_0.1.14            edgeR_3.40.0          
[131] rjson_0.2.21           lifecycle_1.0.3        nlme_3.1-160           jsonlite_1.8.3         viridisLite_0.4.1     
[136] limma_3.54.0           fansi_1.0.3            pillar_1.8.1           lattice_0.20-45        fastmap_1.1.0         
[141] httr_1.4.4             survival_3.4-0         glue_1.6.2             png_0.1-7              iterators_1.0.14      
[146] stringi_1.7.8          sass_0.4.2             textshaping_0.3.6      memoise_2.0.1          dplyr_1.0.10          
[151] irlba_2.3.5.1          future.apply_1.10.0

**Comment by kevinrue** [comment]
Not sure whether you the user or us the developers need to define a method for the generic `.showSelectionDetails`. I guess you can define one yourself as a hot fix, and I'll look into this ASAP to see if we need a fix in the package itself. I haven't use custom tables recently and can't remember how it's meant to be used.

**Comment by LTLA** [comment]
Probably this is a job for `createCustomTable`. The best solution would be to add an argument to `createCustomTable` that allows the definer to specify whether they want to do something special upon a selection of a table row. The default can probably just be `NULL`, but if it isn't, the definer can show, e.g., links to custom resources.

```
# inside createCustomTable
setMethod(".showSelectionDetails", className, function(x) {
    if (!is.null(user.supplied.function)) {
        user.supplied.function(slot(x, .TableSelected))
    }
})
```

We must have added `.showSelectionDetails` some time after `createCustomTable`. Could throw in some tests checking that the latter defines methods for the former.

**Comment by kevinrue** [comment]
Fixed by #617

So sorry: I had the PR ready back in February, and personal events distracted me from it and I forgot to merge. Done now!
